### PR TITLE
Clojure Guest Post

### PR DESCRIPTION
**Pull request by mmzsource**
Kind request to merge and publish the clojure post :)

### CODE REVIEW

File: _posts/2017-06-26-bezier-in-clojure.md [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+	- `xs` are the all x values resulting from applying the `b3x` anonymous function with all the range values.
+	- ~~what the `ys` are is left as an assignment for the curious reader~~ `ys` are the all y values resulting from applying the `b3y` anonymous function with all the range values.
+	- incanter can work with columns similarly to spreadsheets. `raw-dataset` is an incanter dataset where 2 columns are `conj[oined]`.
+	- Default, these columns are called `col-0` and `col-1` respectively, so in `dataset` these are renamed to `:x` and `:y`
+	- xy-plot contains an incanter chart where `dataset` provides the data, the x-axis - and y-axis values are found in columns `:x` and `:y` respectively and viewing the points is set to `true`
+- and make it return an `xy-plot`
+
+
+## I love it when a plan comes together
+
```
Comment (npalm) [inline _posts/2017-06-26-bezier-in-clojure.md]: Maybe add here some more details, and explain the curve will be opened in an external java window.

File: _posts/2017-06-26-bezier-in-clojure.md [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+- define a function named `view-bezier-plot`
+- let it have 3 arguments which are destructured into their 2D x & y coordinates 
+- let the fourth argument be the title of the plot
+- `let` it have some local variables:
+	- `b3x` takes all x coordinates of the 3 points
+	- `b3y` takes the y coordinates
+	- `xs` are the all x values resulting from applying the `b3x` anonymous function with all the range values.
+	- ~~what the `ys` are is left as an assignment for the curious reader~~ `ys` are the all y values resulting from applying the `b3y` anonymous function with all the range values.
+	- incanter can work with columns similarly to spreadsheets. `raw-dataset` is an incanter dataset where 2 columns are `conj[oined]`.
```
Comment (npalm) [inline _posts/2017-06-26-bezier-in-clojure.md]: conj[oined] ??

File: _posts/2017-06-26-bezier-in-clojure.md [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+This function basically states:
+
+- define a function named `view-bezier-plot`
+- let it have 3 arguments which are destructured into their 2D x & y coordinates 
+- let the fourth argument be the title of the plot
+- `let` it have some local variables:
+	- `b3x` takes all x coordinates of the 3 points
+	- `b3y` takes the y coordinates
+	- `xs` are the all x values resulting from applying the `b3x` anonymous function with all the range values.
+	- ~~what the `ys` are is left as an assignment for the curious reader~~ `ys` are the all y values resulting from applying the `b3y` anonymous function with all the range values.
```
Comment (npalm) [inline _posts/2017-06-26-bezier-in-clojure.md]: I don't get the strike through

File: _posts/2017-06-26-bezier-in-clojure.md [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+```clojure
+(require '[incanter.core   :as incanter])
+(require '[incanter.charts :as charts])
+```
+
+Nothing fancy going on here. Simply 'import' statements the `clojure` way.
+
+## Helper function
+
+I decided to hand-roll my own `pow` function:
```
Comment (npalm) [inline _posts/2017-06-26-bezier-in-clojure.md]: Maybe you can mention here that you will use the function later

File: _posts/2017-06-26-bezier-in-clojure.md [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+author:     "Maarten Metz"
+header-img: "img/htc.jpg"
+tags:       [clojure, bézier, functional, incanter]
+---
+
+For reasons I might explain in another post, I was reading about [Bézier curves on wikipedia](https://en.wikipedia.org/wiki/Bézier_curve) the other day. Especially the paragraph on "Constructing Bézier curves" made me understand intuitively how these curves are constructed. I wondered if `clojure` could help me easily visualise some of these curves so I fired up a [REPL](https://en.wikipedia.org/wiki/Read–eval–print_loop) (Read-Eval-Print-Loop) to investigate.
+
+## Setup
+
+I assume you have a working [leiningen](https://leiningen.org) setup with the [lein-try plugin](https://github.com/rkneufeld/lein-try) installed.
```
Comment (npalm) [inline _posts/2017-06-26-bezier-in-clojure.md]: I had some issues with installing lein-try. I have installed leiningen via brew and added the plugin as the suggested on the github repo. But only the second suggestion works for me:

Not working
;; All alone:
[lein-try "0.4.3"]

Works
;; The whole thing:
{:user {:plugins [[lein-try "0.4.3"]]}}

Comment (mmzsource) [inline _posts/2017-06-26-bezier-in-clojure.md]: Yeah, if you're used to the .lein settings file you will 'get' what "all alone" means. However, I want to stay far away from install issues on all different platform and therefore only point to the library site.

File: _posts/2017-06-26-bezier-in-clojure.md (hunk 4 above)
Comment (mmzsource) [inline _posts/2017-06-26-bezier-in-clojure.md]: Good point

File: _posts/2017-06-26-bezier-in-clojure.md (hunk 3 above)
Comment (mmzsource) [inline _posts/2017-06-26-bezier-in-clojure.md]: Will remove it

File: _posts/2017-06-26-bezier-in-clojure.md (hunk 2 above)
Comment (mmzsource) [inline _posts/2017-06-26-bezier-in-clojure.md]: conjoined: brought together so as to meet, touch, overlap, or unite
There is even a clojure event called clojure/conj. The conj function existed long before we were born. It's available in all lisps I have seen.

File: _posts/2017-06-26-bezier-in-clojure.md (hunk 1 above)
Comment (mmzsource) [inline _posts/2017-06-26-bezier-in-clojure.md]: good point

**Pull request merged**